src/QsCompiler/QirGeneration/CompilationSteps.cs

```diff
@@ -46,9 +46,17 @@ internal static bool VerifyGenerationPrecondition(QsCompilation compilation, Lis
             }
         }
 
-        internal static Generator CreateAndPopulateGenerator(QsCompilation compilation, TargetCapability? targetCapability)
+        internal static Generator CreateAndPopulateGenerator(QsCompilation compilation, IDictionary<string, string?> assemblyConstants)
         {
+            var targetCapability = assemblyConstants.TryGetValue(ReservedKeywords.AssemblyConstants.TargetCapability, out var capability) && !string.IsNullOrWhiteSpace(capability)
+                ? TargetCapability.TryParse(capability) // null if parsing fails
+                : null;
+            var isMicrosoftSimulator =
+                assemblyConstants.TryGetValue(ReservedKeywords.AssemblyConstants.ProcessorArchitecture, out var architecture)
+                && architecture == ReservedKeywords.AssemblyConstants.MicrosoftSimulator;
+
             compilation =
+                isMicrosoftSimulator ||
```

Comment:
I believe we might now be effectively applying `AddOutputRecording` for all cases where we generate QIR. This condition could go away and the transformation could always be applied.

Reply:
Should I also get rid of the part of the condition pertaining to the target capabilities?

Reply:
Yes, it seems like you already did that :)